app/uk/gov/hmrc/pillar2externalteststub/controllers/SubscriptionController.scala

```diff
@@ -34,25 +32,9 @@ class SubscriptionController @Inject() (
 ) extends BackendController(cc)
     with Logging {
 
-  def retrieveSubscription(plrReference: String): Action[AnyContent] =
-    (Action andThen authFilter).async { implicit request =>
-      logger.info(s"Retrieving subscription for PLR reference: $plrReference")
-
-      plrReference match {
-        case "XEPLR0123456404" =>
-          Future.successful(NotFound(Json.toJson(NotFoundSubscription.response)))
-        case "XEPLR0123456500" =>
-          Future.successful(InternalServerError(Json.toJson(ServerError500.response)))
-        case "XEPLR0123456503" =>
-          Future.successful(ServiceUnavailable(Json.toJson(ServiceUnavailable503.response)))
-        case "XEPLR5555555555" =>
-          Future.successful(Ok(Json.toJson(SubscriptionSuccessResponse.successfulDomesticOnlyResponse(plrReference))))
-        case "XEPLR1234567890" =>
-          Future.successful(Ok(Json.toJson(SubscriptionSuccessResponse.successfulNonDomesticResponse(plrReference))))
-        case "XEPLR0987654321" =>
-          Future.successful(Ok(Json.toJson(NilReturnSuccess.successfulResponse)))
```

Comment:
Also documented in the README, I'm unsure why a NilReturnSuccess is being returned from this controller and not SubmitUKTRController, isn't the Nil Return a subtype of UKTR?

The file seems to belong to neither package now.

<img width="345" alt="Screenshot 2024-12-15 at 19 46 22" src="https://github.com/user-attachments/assets/91df7ad7-5ea1-4803-96ce-d757936f793c" />

Reply:
Yes, a good spot. 
It looks like this case class should belong in the response package :  
/pillar2-external-test-stub/app/uk/gov/hmrc/pillar2externalteststub/models/uktr/response
I can refactor this as part of : PIL-1297 Validation of MNE/UK UKTR Nil Return [which I am working on finishing today].